docs/rules/azurerm_app_service_app_insights_hidden_link.md

```diff
@@ -0,0 +1,127 @@
+resource "azurerm_windows_function_app" "example" {
+  app_settings = {
+    "APPINSIGHTS_INSTRUMENTATIONKEY" = "example-key"
+  }
+}
+
+# Non-compliant: Linux Function App with Application Insights in site_config without ignore_changes
+resource "azurerm_linux_function_app" "example" {
+  site_config {
+    application_insights_connection_string = "example-connection"
+  }
+}
+
+# Compliant: Linux Function App with proper ignore_changes
+resource "azurerm_linux_function_app" "example" {
+  app_settings = {
+    "APPLICATIONINSIGHTS_CONNECTION_STRING" = "example"
+  }
+  
+  lifecycle {
+    ignore_changes = [
+      tags["hidden-link: /app-insights-conn-string"],
+      tags["hidden-link: /app-insights-instrumentation-key"],
+      tags["hidden-link: /app-insights-resource-id"],
+    ]
+  }
+}
+
+# Compliant: Windows Web App Slot with proper ignore_changes
+resource "azurerm_windows_web_app_slot" "example" {
+  app_settings = {
+    "APPINSIGHTS_INSTRUMENTATIONKEY" = "example"
+  }
+  
+  lifecycle {
+    ignore_changes = [
+      tags["hidden-link: /app-insights-conn-string"],
+      tags["hidden-link: /app-insights-instrumentation-key"],
+      tags["hidden-link: /app-insights-resource-id"],
+    ]
+  }
+}
+
+# Compliant: Windows Function App with Application Insights in site_config and proper ignore_changes
+resource "azurerm_windows_function_app" "example" {
+  site_config {
+    application_insights_key = "example-key"
+  }
+  
+  lifecycle {
+    ignore_changes = [
+      "tags["hidden-link: /app-insights-conn-string"]",
+      "tags["hidden-link: /app-insights-instrumentation-key"]",
+      "tags["hidden-link: /app-insights-resource-id"]",
+    ]
+  }
+}
+```
+
+## Why
+
+When Application Insights is configured for Azure App Service resources (Web Apps, Web App Slots, Function Apps, or Function App Slots for both Linux and Windows), Azure automatically adds hidden-link tags to the resource. These tags can change during deployments and may cause unnecessary Terraform diffs. Using `lifecycle { ignore_changes }` for these specific tags prevents Terraform from attempting to manage these Azure-managed tags.
+
+Application Insights can be configured in two ways:
+1. **app_settings**: Using `APPLICATIONINSIGHTS_CONNECTION_STRING` or `APPINSIGHTS_INSTRUMENTATIONKEY` keys
+2. **site_config**: Using `application_insights_connection_string` or `application_insights_key` attributes (only applicable to Function Apps)
+
+This rule checks for the presence of Application Insights configuration in both locations and ensures that the corresponding hidden-link tags are properly ignored.
+
+## How to Fix
+
+Add a `lifecycle` block with `ignore_changes` containing the Application Insights hidden-link tags:
+
+```hcl
+lifecycle {
+  ignore_changes = [
+    "tags["hidden-link: /app-insights-conn-string"]",
+    "tags["hidden-link: /app-insights-instrumentation-key"]", 
+    "tags["hidden-link: /app-insights-resource-id"]",
```

Comment:
```suggestion
    tags["hidden-link: /app-insights-conn-string"],
    tags["hidden-link: /app-insights-instrumentation-key"], 
    tags["hidden-link: /app-insights-resource-id"],
```

By the way, there is also the following legacy syntax, but it is up to you whether to support it or not.

```hcl
ignore_changes = [
  "tags['hidden-link: /app-insights-conn-string']"
]
```